### ISSUE TITLE
Raising validation errors for multiple field names in schema-scoped validators

### ISSUE THREAD

**Issue by ksesong** [marshmallow-code/marshmallow#118]
Supposing a schema where only one field is required (i.e. all the others must be missing), I am using a schema validator to check this condition, as I need the state of the other fields. If there is no such non-null field, I want to raise an error for each field (as they are equally responsible). If I however raise the error, it stalls the whole process. I could raise a number of ValidationError via a list, but I can't assign these errors to different field names.

``` python
class only_one_field_present(object):
    def __init__(self, choices):
        self.choices = choices
    def run(self, schema, data):
        if not any([data[name] for name in self.choices]):
            # -- Raise ValidationError for all the fields names in choices...

class DummySchema(Schema):
    __validators__ = [only_one_field_present(['field_a', 'field_b']).run]
    field_a = fields.Field()
    field_b = fields.Field()
```

This is what I would like to get in the end:

``` python
errors = {
    'field_a': ['one_required'],
    'field_b': ['one_required']
}
```

**Comment by sloria** [comment]
I suppose we could allow multiple fields to be passed to a validation error.

``` python
error_fields = #... compute which fields failed validation
raise ValidationError('one_required', error_fields)
```

Would this meet your use case?

**Comment by ksesong** [comment]
Yes.

**Comment by sloria** [comment]
OK, this feature has been merged into `dev`. 

The feature introduced a backwards-incompatible change to the `UnmarshallingError` interface, so this will probably be released in version 2.0.

You can still install the `dev` branch with:

``` python
pip install -U git+https://github.com/marshmallow-code/marshmallow.git@dev
```
